File: GeoServices/Logger/ServiceEventWriter.vb
Namespace Logger
    ''' <summary>
    ''' Provee la interfaz para el event viewer
    ''' </summary>
    ''' <remarks>Se debe especificar el nombre de la aplicación en Config.xml</remarks>
    Class ServiceEventWriter
        Private Shared _source As String = XML.XMLLoggerGetter.getName() & "Events"
        Private Shared _log As String = XML.XMLLoggerGetter.getName() & "Log"

        Private Shared Function checkCreate() As EventLog
            If Not EventLog.SourceExists(_source) Then EventLog.CreateEventSource(_source, _log)
            Return New EventLog(_log, System.Environment.MachineName, _source)
        End Function

        Public Shared Sub Warn(ByVal message As Object)
            checkCreate().WriteEntry(message.ToString(), EventLogEntryType.Warning)
        End Sub

        Public Shared Sub [Error](ByVal message As Object)
            checkCreate().WriteEntry(message.ToString(), EventLogEntryType.Error)
        End Sub

        Public Shared Sub Info(ByVal message As Object)
            checkCreate().WriteEntry(message.ToString(), EventLogEntryType.Information)
        End Sub
    End Class
End Namespace

File: GeoServicesSharp/Logger/ServiceEventWriter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Diagnostics;

namespace GeoServicesSharp.Logger
{
    /// <summary>
    /// Provee la interfaz para el event viewer
    /// </summary>
    /// <remarks>Se debe especificar el nombre de la aplicación en Config.xml</remarks>
    class ServiceEventWriter
    {
        private static string _source = XML.XMLLoggerGetter.GetName() + "Events";
        private static string _log = XML.XMLLoggerGetter.GetName() + "Log";

        private static EventLog checkCreate()
        {
            if (!EventLog.SourceExists(_source)) EventLog.CreateEventSource(_source, _log);
            return new EventLog(_log, System.Environment.MachineName, _source);
        }

        public static void Warn(object message)
        {
            checkCreate().WriteEntry(message.ToString(), EventLogEntryType.Warning);
        }

        public static void Error(object message)
        {
            checkCreate().WriteEntry(message.ToString(), EventLogEntryType.Error);
        }

        public static void Info(object message)
        {
            checkCreate().WriteEntry(message.ToString(), EventLogEntryType.Information);
        }
    }
}